src/postgres/commands/createPostgresServer/steps/PostgresServerNameStep.ts

```diff
@@ -46,13 +47,10 @@ async function validatePostgresServerName(name: string, client: PostgreSQLManage
         return localize('serverNamePrefixSuffixCheck', 'Server name must not start or end in a hyphen.');
     }
 
-    const availabilityRequest: NameAvailabilityRequest = { name: name, type: "Microsoft.DBforPostgreSQL" };
-    const availability: NameAvailability = (await client.checkNameAvailability.execute(availabilityRequest));
+    const availability: AbstractNameAvailability = await client.checkNameAvailability.execute({name: name, type: "Microsoft.DBforPostgreSQL"});
```

Comment:
This isn't workin for me for flexible servers. It's silently failing under the covers because of this error:

> Unknown resource type specified for a subscription level name availability check. Supported resource types are: Microsoft.DBforPostgreSQL/flexibleServers.